cpp/tests/wrappers/bools_test.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/wrappers/bool.hpp>
+
+#include <tests/utilities/cudf_gtest.hpp>
+
+#include <random>
+
+struct Bool8Test : public ::testing::Test {
+
+  Bool8Test()
+      : dist{std::numeric_limits<uint8_t>::lowest(),
+             std::numeric_limits<uint8_t>::max()} {
+
+    // Use constant seed for deterministic results
+    rng.seed(0);
+  }
+
+  std::mt19937 rng;
+  std::uniform_int_distribution<uint8_t> dist;
+
+  uint8_t rand(){
+      return dist(rng);
+  }
+};
+
+template <typename SourceType>
+struct Bool8CtorTest : public ::testing::Test {
+  Bool8CtorTest() {
+    // Use constant seed for deterministic results
+    rng.seed(0);
+  }
+
+  std::mt19937 rng;
+
+  using uniform_distribution = typename std::conditional_t<
+      std::is_same<SourceType, bool>::value, std::bernoulli_distribution,
+      std::conditional_t<std::is_floating_point<SourceType>::value,
+                         std::uniform_real_distribution<SourceType>,
+                         std::uniform_int_distribution<SourceType>>>;
+
+  uniform_distribution dist{};
+
+  SourceType rand(){
+      return dist(rng);
+  }
+};
+
+/**
+ * @brief The number of test trials
+ */
+static constexpr int NUM_TRIALS{10000};
+
+using Bool8CastSourceTypes = ::testing::Types<char, bool, float, double, int32_t, int64_t>;
```

Comment:
```suggestion
using Bool8CastSourceTypes = Concat<cudf::test::NumericTypes, bool>;
```

Reply:
Fixed

Reply:
Actually I forgot that `bool` _is_ numeric, so there's no need for the `concat`.

Reply:
Adding bool to the list of numerictypes causes compiler errors for the `fixed_width_column_wrapper_tests` on [this line](https://github.com/rapidsai/cudf/blob/3253ec2bea8245584f707f87fba4036c80f80b20/cpp/tests/utilities_tests/column_wrapper_tests.cu#L59) -- seems like it doesn't want to convert arbitrary numbers to booleans:

```
../tests/utilities/column_wrapper.hpp(111): error: no instance of constructor "rmm::device_buffer::device_buffer" matches the argument list
            argument types are: (void, unsigned long)
          detected during:
            instantiation of "rmm::device_buffer cudf::test::detail::make_elements<Element,InputIterator>(InputIterator, InputIterator) [with Element=__nv_bool, InputIterator=thrust::transform_iterator<lambda [](auto)->auto, thrust::counting_iterator<cudf::size_type, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::use_default, thrust::use_default>]" 
(223): here
            instantiation of "cudf::test::fixed_width_column_wrapper<Element>::fixed_width_column_wrapper(InputIterator, InputIterator) [with Element=__nv_bool, InputIterator=thrust::transform_iterator<lambda [](auto)->auto, thrust::counting_iterator<cudf::size_type, thrust::use_default, thrust::use_default, thrust::use_default>, thrust::use_default, thrust::use_default>]" 
../tests/utilities_tests/column_wrapper_tests.cu(48): here
            instantiation of "void FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<gtest_TypeParam_>::TestBody() [with gtest_TypeParam_=__nv_bool]" 
googletest/install/include/gtest/internal/gtest-internal.h(484): here
            implicit generation of "FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<gtest_TypeParam_>::~FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test() [with gtest_TypeParam_=__nv_bool]" 
googletest/install/include/gtest/internal/gtest-internal.h(484): here
            instantiation of class "FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<gtest_TypeParam_> [with gtest_TypeParam_=__nv_bool]" 
googletest/install/include/gtest/internal/gtest-internal.h(484): here
            [ 3 instantiation contexts not shown ]
            implicit generation of "testing::internal::TestFactoryImpl<TestClass>::~TestFactoryImpl() [with TestClass=FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<__nv_bool>]" 
googletest/install/include/gtest/internal/gtest-internal.h(649): here
            instantiation of class "testing::internal::TestFactoryImpl<TestClass> [with TestClass=FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<__nv_bool>]" 
googletest/install/include/gtest/internal/gtest-internal.h(649): here
            implicit generation of "testing::internal::TestFactoryImpl<TestClass>::TestFactoryImpl() [with TestClass=FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<__nv_bool>]" 
googletest/install/include/gtest/internal/gtest-internal.h(649): here
            instantiation of class "testing::internal::TestFactoryImpl<TestClass> [with TestClass=FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test<__nv_bool>]" 
googletest/install/include/gtest/internal/gtest-internal.h(649): here
            instantiation of "__nv_bool testing::internal::TypeParameterizedTest<Fixture, TestSel, Types>::Register(const char *, testing::internal::CodeLocation, const char *, const char *, int) [with Fixture=FixedWidthColumnWrapperTest, TestSel=testing::internal::TemplateSel<FixedWidthColumnWrapperTest_NonNullableIteratorConstructor_Test>, Types=gtest_type_params_FixedWidthColumnWrapperTest_]" 
../tests/utilities_tests/column_wrapper_tests.cu(41): here
```